consensus/src/chained_bft/event_processor.rs

```diff
@@ -420,12 +420,14 @@ impl<T: Payload> EventProcessor<T> {
             .committed_block_id()
             .and_then(|new_commit| self.block_store.get_block(new_commit))
         {
-            // We don't want to use NIL commits for pacemaker round interval calculations.
-            if !block.is_nil_block() {
-                highest_committed_proposal_round = Some(block.round());
+            if block.round() > self.block_store.root().round() {
```

Comment:
seems like we could potentially unifying the restarting with the playing of a QC